### PR TITLE
Set CONDA_BLD_PATH when using pixi during osx build

### PR DESCRIPTION
**Pull request by hadim**
Checklist
* [x] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

This is an attempt to fix https://github.com/conda-forge/conda-smithy/issues/2145. You can see it in action at https://github.com/conda-forge/docling-parse-feedstock/pull/3

Note the fix is a simple and naive patch but looks fragile to me. Please let me know if you can think of another solution.

### CODE REVIEW

**Review by beckermr — DISMISSED** [review]
I wonder if we should set this on all platforms?

Also, the code that finds the outputs is in the ci-setup feedstock. We need to consult that code to ensure what we are doing here works with it.

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 1]
```diff
@@ -9,6 +9,15 @@ set -xe
 MINIFORGE_HOME=${MINIFORGE_HOME:-${HOME}/miniforge3}
 MINIFORGE_HOME=${MINIFORGE_HOME%/} # remove trailing slash
```
Comment (jaimergp) [inline conda_smithy/templates/run_osx_build.sh.tmpl:10]: I think we can include [this blob](https://github.com/conda-forge/staged-recipes/blob/778e1c767ae07a9ba9eecc60e927325ce05f93d3/.scripts/run_osx_build.sh#L9-L11) from `staged-recipes` here instead.

**Comment by jaimergp** [comment]
> I wonder if we should set this on all platforms?

Yea I think it wouldn't hurt. We are doing it in staged-recipes.

**Comment by jaimergp** [comment]
Testing at https://github.com/conda-forge/docling-parse-feedstock/pull/5

**Comment by beckermr** [comment]
I wonder why we set it in ci setup for win, but in smithy for Unix. I don't fully understand when stuff goes in one place versus the other. We should maybe clarify in future prs.

**Pull request merged**